### PR TITLE
Refactor: `MutList` to use `Iterable`

### PR DESCRIPTION
**Pull request by ninaandrup**
Related to #2999

### CODE REVIEW

File: main/src/library/MutList.flix [hunk 1]
```diff
@@ -566,19 +566,20 @@ namespace MutList {
...
-        let len2 = deref l2;
-        if (len2 > Array.length(deref a1) - len1)
-            reserve!(len2, v1)
-        else
-            ();
-        Array.updateSequence!(len1, deref a2, deref a1);
-        l1 := len1 + len2
+    /// Inserts all the elements in the collection `t` into the mutable list `v`.
+    ///
+    pub def append!(v: MutList[a], t: t[a]): Unit & Impure with Iterable[t] =
```
Comment (ninaandrup) [inline main/src/library/MutList.flix:571]: Should `v` be the last argument?

Comment (magnus-madsen) [reply main/src/library/MutList.flix:571]: Maybe not sure, can you open a new ticket?

File: main/src/library/MutList.flix [hunk 2]
```diff
@@ -566,19 +566,20 @@ namespace MutList {
...
+        let iter = Iterable.iterator(t);
+        def loop(a, b) = {
```
Comment (magnus-madsen) [inline main/src/library/MutList.flix:573]: I think you should just call `Iterator.foreach(x -> MutList.push!(....))`

Comment (magnus-madsen) [reply main/src/library/MutList.flix:573]: :)

Comment (ninaandrup) [reply main/src/library/MutList.flix:573]: Ah, yes, you are right :)

File: main/src/library/MutList.flix (hunk 1 above)
Comment (ninaandrup) [reply main/src/library/MutList.flix:571]: Yes, I will do that

**Pull request merged**